packages/constellation/src/components/Compositions/Switcher/WalletSwitcher.tsx

```diff
@@ -0,0 +1,47 @@
+import cx from 'classnames'
+import React, { forwardRef, useMemo } from 'react'
+
+import { Text } from '../../Base'
+import { WalletComponent as ComponentType, WalletProps } from './Switcher.types'
+
+const Switcher: ComponentType = forwardRef(
+  <T extends React.ElementType = 'button'>(
+    { as, disabled = false, id, status, ticker, type, ...otherProps }: WalletProps<T>,
+    ref?: PolymorphicRef<T>,
+  ) => {
+    const Component = as || 'button'
+    const fallbackType = Component === 'button' && !type ? 'button' : undefined
+
+    const statusColor = {
+      error: 'bg-error',
+      success: 'bg-success',
+      warning: 'bg-warning',
+    }
+
+    const truncatedId = useMemo(() => {
```

Comment:
could you make this accept a parameter for maxLength, and pass it to utils?

Reply:
Done!